app/UpscanStubModule.scala

```diff
@@ -1,9 +1,20 @@
+import akka.actor.ActorSystem
+import javax.inject.{Inject, Provider, Singleton}
 import play.api.{Configuration, Environment}
 import play.api.inject.{Binding, Module}
-import services.{HttpNotificationService, NotificationService}
+import services.{HttpNotificationSender, NotificationQueueProcessor, NotificationSender}
 
 class UpscanStubModule extends Module {
   override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = Seq(
-    bind[NotificationService].to[HttpNotificationService]
+    bind[NotificationSender].to[HttpNotificationSender],
+    bind[NotificationQueueProcessor].toProvider[NotificationQueueProcessorProvider]
   )
+
+}
+
+@Singleton
```

Comment:
Didn't we have a problem making a Provider a Singleton previously in upscan-listener?